tee-worker/omni-executor/ts-tests/stress-tests/src/dashboard/pages/api/sessions/index.ts

```diff
@@ -0,0 +1,103 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import fs from 'fs';
+import path from 'path';
+
+interface SessionSummary {
+  sessionId: string;
+  startTime: number;
+  endTime: number;
+  duration: number;
+  status: string;
+  config: any;
+  maxSustainableQPS: number;
+  recommendedMaxQPS: number;
+  overallStats: {
+    totalRequests: number;
+    totalSuccessful: number;
+    totalFailed: number;
+    overallSuccessRate: number;
+    avgResponseTime: number;
+    maxResponseTime: number;
+    minResponseTime: number;
+  };
+}
+
+const RESULT_PATH = '/Users/verin/Repo/Heima/heima/tee-worker/omni-executor/ts-tests/stress-tests/stress-test-results';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method !== 'GET') {
+    return res.status(405).json({ error: 'Method not allowed' });
+  }
+
+  try {
+    // Check if results directory exists
+    if (!fs.existsSync(RESULT_PATH)) {
+      return res.json([]);
+    }
+
+    const sessions: SessionSummary[] = [];
+    const sessionDirs = fs.readdirSync(RESULT_PATH)
+      .filter(dir => {
+        const fullPath = path.join(RESULT_PATH, dir);
+        return fs.statSync(fullPath).isDirectory() && dir.startsWith('stress_test_');
+      });
+
+    for (const sessionDir of sessionDirs) {
+      const sessionPath = path.join(RESULT_PATH, sessionDir);
+      const summaryPath = path.join(sessionPath, 'session_summary.json');
+
+      if (fs.existsSync(summaryPath)) {
+        try {
+          const summaryContent = fs.readFileSync(summaryPath, 'utf8');
+          const summary = JSON.parse(summaryContent) as SessionSummary;
+          sessions.push(summary);
+        } catch (err) {
+          console.error(`Error reading session summary for ${sessionDir}:`, err);
+        }
+      } else {
+        // Try to create session_summary from session-result.json if it exists
+        const resultPath = path.join(sessionPath, 'session-result.json');
+        if (fs.existsSync(resultPath)) {
+          try {
+            const resultContent = fs.readFileSync(resultPath, 'utf8');
```

Comment:
would it better to use some local db like sqlite?

Reply:
We may be running on different machines simultaneously, and the local database can only be stored locally. The advantage of using JSON is that when we analyze the data for the dashboard, we can simply combine the JSON data from different machines.

Reply:
I'm not sure I follow, this is reading from disk. And what would be your goal running this from multiple machines? Running these tests in parallel would be enough. Anyways, it's not a big deal, the format used should not affect the results

Reply:
>what would be your goal running this from multiple machines?
 My original idea was to have multiple team members run stress tests simultaneously from different machines , then collect all the test results and aggregate them on dashboard page.